### PR TITLE
Entities and Relations

### PR DESCRIPTION
**Pull request by pradiprach**

### CODE REVIEW

File: src/main/java/com/shopping/entities/Account.java [hunk 1]
```diff
@@ -22,6 +22,17 @@
 	
 	@Column
 	private Date close;
+	
+	@OneToOne
+	private ShoppingCart shoppingCart;
```
Comment (kbsaxena) [inline src/main/java/com/shopping/entities/Account.java:41]: Add relationship with Customer

File: src/main/java/com/shopping/entities/Customer.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	@Column
+	private String email;
+	
+	@Column
+	private String fullName;
+	
+	@Column
+	private Date dob;
+	
+	@OneToMany
```
Comment (kbsaxena) [inline src/main/java/com/shopping/entities/Customer.java:28]: Add Bidirectional relationship to Account

File: src/main/java/com/shopping/entities/Customer.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	private List<Address> address;
+	
+	@OneToMany
+	private List<Phone> phoneNumbers;
```
Comment (kbsaxena) [inline src/main/java/com/shopping/entities/Customer.java]: This should be a Set.All phone numbers will be unique

File: src/main/java/com/shopping/entities/Phone.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Table;
+
+@Entity
+@Table(name="mobile")
+public class Phone extends BaseEntity {
+	
+	@Column
+	private String mobileNumber;
```
Comment (kbsaxena) [inline src/main/java/com/shopping/entities/Phone.java:13]: Add getter and setters

File: src/main/java/com/shopping/entities/Phone.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+	
+	@Column
+	private String countryCode;
+}
```
Comment (kbsaxena) [inline src/main/java/com/shopping/entities/Phone.java:44]: Add Bidirectional relationship to Customer

File: src/main/java/com/shopping/entities/ShoppingCart.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+	private Date created;
+	
+	@OneToOne
+	private User user;
+	
+	@OneToOne
+	private Account account;
+	
+	@OneToMany
+	private TreeSet<LineItem> lineItems;
```
Comment (kbsaxena) [inline src/main/java/com/shopping/entities/ShoppingCart.java]: we need to initialize here only with new object to eradicate null pointers

File: src/main/java/com/shopping/entities/User.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+	@Column
+	private String password;
+	
+	@Column
+	private UserState userState;
+	
+	@OneToOne
+	private Customer customer;
+	
+	@OneToOne(optional=true)
```
Comment (kbsaxena) [inline src/main/java/com/shopping/entities/User.java:24]: remove optional parameter

**Review by kbsaxena — CHANGES_REQUESTED** [review]

File: src/main/java/com/shopping/entities/User.java (hunk 7 above)
Comment (pradiprach) [reply src/main/java/com/shopping/entities/User.java:24]: Please verify the image. There it is specified as either 0 or 1. That is the reason I marked it optional.

Comment (kbsaxena) [reply src/main/java/com/shopping/entities/User.java:24]: Ok

**Pull request closed without merging**